src/app/post-page/post-page.component.ts

```diff
@@ -62,34 +115,90 @@ export class PostPageComponent implements OnInit {
     },
     (err) => {
       // fill in this part...
+      this._router.navigate(['/pageNotFound']); // ?
       console.log(err);
     })
   }
+  
+  // Not working
+  // checkAnonymous() {
+  //   const name = this.commentForm.get('name');
+  //   this.isAnonymous = true;
+  //   this.isReveal = false;
+  //   name.clearValidators();
+  //   name.setValue('');
+  //   name.disable();
+  //   name.updateValueAndValidity();
+  // }
+
+  // checkReveal() {
+  //   const name = this.commentForm.get('name');
+  //   this.isReveal = true;
+  //   this.isAnonymous = false;
+  //   name.setValidators(Validators.required);
+  //   name.enable();
+  //   name.updateValueAndValidity();
+  // }
+
+  // checkRevealAlt() {
+  //   const name = this.commentForm.get('name');
+  //   this.isReveal = true;
+  //   this.isAnonymous = false;
+  //   name.setValidators(Validators.required);
+  //   name.enable();
+  //   name.updateValueAndValidity();
+  // }
 
   goPrevious() {
-    let previousId = this.post.id - 1;
-    this._router.navigate(['home/post'], { queryParams: { id: previousId } }); // is the route correct?
+    let previousId = (this.post.id - 1) <= 0 ? this.numberOfApprovedPosts : this.post.id - 1;
+    // this._router.navigate(['home/post'], { queryParams: { id: previousId } });
+    this._router.navigate(['./'], 
+                          { relativeTo: this._route, 
+                            queryParams: { id: previousId }}); // relative navigation
   }
 
   goNext() {
-    let nextId = this.post.id + 1;
-    this._router.navigate(['home/post'], { queryParams: { id: nextId } }); // is the route correct?
+    let nextId = (this.post.id + 1) >= this.numberOfApprovedPosts ? 1 : this.post.id + 1;
+    // this._router.navigate(['home/post'], { queryParams: { id: nextId } });
+    this._router.navigate(['./'], 
+                          { relativeTo: this._route, 
+                            queryParams: { id: nextId }}); // relative navigation
   }
 
   updatePostLikes() {
-    this._postService.updateLikes(this.post.id, this.post.likes + 1).subscribe(); // subscribe need para?
+    this._postService.updateLikes(this.post.id, this.post.likes + 1)
+      .subscribe(() => { this.reloadData; });
   }
 
   updateCommentLikes(comment: Comment) {
-    this._commentService.updateLikes(comment.id, comment.likes + 1).subscribe(); // subscribe need para?
+    this._commentService.updateLikes(comment.id, comment.likes + 1)
+      .subscribe(() => { this.reloadData; });
   }
 
-  createNewComment(text: string) {
+  createNewComment(text: string, poster: string) {
     const newComment = {
+      id: this.comments.length + 1,
```

Comment:
warning: the comments sent from the backend are paginated, so that the website wont get slowed down when there are a lot of comments for just one single post. also there's a reason why i omitted so many details while creating the comment object as backend will help me fill in the details that i omitted (i have set up the default values at the backend, did u clone the backend repo?).